Add hints

Sveteeen opened the pull request:
## Изменения
<!-- Добавила информационные хинты к каждой выплате в обратном калькуляторе, которые сообщают почему поступают именно такие выплаты. -->

## Детали реализации
<!-- Здесь можно описать технические детали по пунктам. 
1. В файл [constants/index.ts](https://github.com/profcomff/calculator-ui/blob/main/src/constants/index.ts) дописала объект TipFromSum
2. В файл (https://github.com/profcomff/calculator-ui/blob/main/src/pages/ReverseCalc.vue)  добавила тултипы в виде серых кнопок с информационным лого-->

## Check-List
<!-- После сохранения у следующих полей появятся галочки, которые нужно проставить мышкой -->
- [x] Вы проверили свой код перед отправкой запроса?
- [x] Вы написали тесты к реализованным функциям?
- [x] Вы не забыли применить форматирование `black` и `isort` для _Back-End_ или `Prettier` для _Front-End_?

Commit:
add TipforSum

Commit:
add grey tooltips

Commit:
add icons

Commit:
change icons

Commit:
add changes

Commit:
change style

Commit:
change style'2

Commit:
change style'2

Commit:
change style'3

BatuevIO commented:
Давай сделаем здесь тоже добавим тултип -- там может быть фиксированный текст, типа "4% от всей полученной суммы". 

![image](https://github.com/user-attachments/assets/0158528c-31b1-4d99-992a-08e76de187e2)

Чтобы так не показывалось, мб имеет смысл сделать disabled если текст пустой

![image](https://github.com/user-attachments/assets/b2ee82ea-c109-41aa-a19a-c6c57f55dac5)

BatuevIO commented on src/pages/ReverseCalc.vue:
```diff
@@ -160,19 +165,55 @@ const formattedStipend = (stipend: number): string => {
+						<div id="gas" class="sum-plus bg-primary">{{ formattedStipend(recount['gas']) }}</div>
+						<v-tooltip :text="TipFromSum['gas'][recount['gas']]">
+							<template #activator="{ props }">
+								<v-btn icon variant="tonal" v-bind="props" class="mt-3">
```
Есть более простой способ добавить иконку в кнопку -- например, как [здесь](https://vuetifyjs.com/en/components/buttons/#icon)

BatuevIO replied:
И тогда не нужны дополнительные зависимости

Sveteeen replied:
Вот тут пробовала раньше сделать как на ссылке, но кнопка перестает вовсе отображаться
Как раз в тг писала раньше об этом, что еще иконка из-за этого не подгружалась, поэтому сделала через зависимость

BatuevIO replied:
Да, видимо, есть какой-то косяк. Давай оставим так тогда

BatuevIO commented on src/pages/ReverseCalc.vue:
```diff
@@ -206,7 +247,7 @@ const formattedStipend = (stipend: number): string => {
 	color: white;
 	border-radius: 999px;
-	margin: 10px 0;
+	margin: 10px 10px;
```
Тут ругается на форматирование. Перед следующей отправкой выполни в терминале ``pnpm check --fix``, он так пофиксит все, что сможет, а остальное выдаст ошибками.

Commit:
add proffee tooltip

Commit:
change tooltips

Sveteeen commented:
> Давай сделаем здесь тоже добавим тултип -- там может быть фиксированный текст, типа "4% от всей полученной суммы".
> 
> ![image](https://private-user-images.githubusercontent.com/124889562/439216037-0158528c-31b1-4d99-992a-08e76de187e2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDY2MTczNTQsIm5iZiI6MTc0NjYxNzA1NCwicGF0aCI6Ii8xMjQ4ODk1NjIvNDM5MjE2MDM3LTAxNTg1MjhjLTMxYjEtNGQ5OS05OTJhLTA4ZTc2ZGUxODdlMi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNTA3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDUwN1QxMTI0MTRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00NTk0YzkwZWRjNWRmZjM4M2MwZGM0YmY5MGMwOGEzNjNiNGY3MWUyYjNmZmFkNzgxOTQ3Mzc4MDJjMDU2ZTI4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.82d4I8MyzjAqG-agh7nk2T98XnW74da1nZx2BITrF44)
> 
> Чтобы так не показывалось, мб имеет смысл сделать disabled если текст пустой
> 
> ![image](https://private-user-images.githubusercontent.com/124889562/439216436-b2ee82ea-c109-41aa-a19a-c6c57f55dac5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDY2MTczNTQsIm5iZiI6MTc0NjYxNzA1NCwicGF0aCI6Ii8xMjQ4ODk1NjIvNDM5MjE2NDM2LWIyZWU4MmVhLWMxMDktNDFhYS1hMTlhLWM2YzU3ZjU1ZGFjNS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNTA3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDUwN1QxMTI0MTRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mODZmM2NlMzllOGI0YWEwZGE0YzBkNGNhOTNiNmEyNjAyNjZmMzJhM2NjNzRjMDM3MzMxMTA3MjFiYjZkYjM0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.O1c_yAe_fTka_FMPRtsY53ol84idlyHTDBIRtfsWXQU)

Доделала

Sveteeen commented:
> Давай сделаем здесь тоже добавим тултип -- там может быть фиксированный текст, типа "4% от всей полученной суммы".
> 
> ![image](https://private-user-images.githubusercontent.com/124889562/439216037-0158528c-31b1-4d99-992a-08e76de187e2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDY2MTczNTQsIm5iZiI6MTc0NjYxNzA1NCwicGF0aCI6Ii8xMjQ4ODk1NjIvNDM5MjE2MDM3LTAxNTg1MjhjLTMxYjEtNGQ5OS05OTJhLTA4ZTc2ZGUxODdlMi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNTA3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDUwN1QxMTI0MTRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00NTk0YzkwZWRjNWRmZjM4M2MwZGM0YmY5MGMwOGEzNjNiNGY3MWUyYjNmZmFkNzgxOTQ3Mzc4MDJjMDU2ZTI4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.82d4I8MyzjAqG-agh7nk2T98XnW74da1nZx2BITrF44)
> 
> Чтобы так не показывалось, мб имеет смысл сделать disabled если текст пустой
> 
> ![image](https://private-user-images.githubusercontent.com/124889562/439216436-b2ee82ea-c109-41aa-a19a-c6c57f55dac5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDY2MTczNTQsIm5iZiI6MTc0NjYxNzA1NCwicGF0aCI6Ii8xMjQ4ODk1NjIvNDM5MjE2NDM2LWIyZWU4MmVhLWMxMDktNDFhYS1hMTlhLWM2YzU3ZjU1ZGFjNS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNTA3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDUwN1QxMTI0MTRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mODZmM2NlMzllOGI0YWEwZGE0YzBkNGNhOTNiNmEyNjAyNjZmMzJhM2NjNzRjMDM3MzMxMTA3MjFiYjZkYjM0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.O1c_yAe_fTka_FMPRtsY53ol84idlyHTDBIRtfsWXQU)

Тут все доделала и поправила

BatuevIO commented on src/pages/ReverseCalc.vue:
```diff
@@ -160,23 +165,83 @@ const formattedStipend = (stipend: number): string => {
-					<div id="gss" class="sum-plus bg-primary">{{ formattedStipend(recount['gss']) }}</div>
+					<div class="d-flex">
+						<div id="gss" class="sum-plus bg-primary">{{ formattedStipend(recount['gss']) }}</div>
+						<v-tooltip
```
Проверил, с мобилки не открывается тултип. Нашел тут, что можно пофиксить, добавив аттрибуты open-on-click, open-on-hover, open-on-focus.

Еще лучше будет, чтобы не дублировать это каждый раз, добавить это в тему vuetify. Для этого в файле /src/vuetify.ts в конфиге (внутри createVuerify) в объекте defaults добавить вот такой объект:

```vue
VTooltip: {
	'open-on-click': true,
	'open-on-hover': true,
	'open-on-focus': true,
}, 
```

Sveteeen replied:
Добавила, в мобилке кнопка у меня отображается, но текст все равно нет...(
Тестила в Devtools

Semedanik replied:
Укажи явно на странничке open-on-click, open-on-hover, open-on-focus, без добавления в vuetify.
Так будет точно все работать)

BatuevIO reviewed (changes requested):
Все супер! Одно последнее изменение, чтобы заработало на мобилке, и можем катить в тест. Дальше уточним про тексты

Commit:
add tooltip in mobile

Commit:
add minor changes

Dmatrushka19 committed:
Сделал определение ширины экрана реактивным через useDisplay;
Добавил обработчик "." и ",", чтобы корректно обрабатывать введенную инфу;
Добавил hasInput с trim() для корректной обработки пустого/пробельного ввода;
использовал тултипы для выведения информации о выплатах (реализовал корректно, в том числе разобрался с иконкой);

Dmatrushka19 committed:
стили

Dmatrushka19 commented:
Последние обновления:

- Добавил реактивное определение ширины: заменил screenWidth = window.innerWidth на useDisplay() и computed isNarrow, чтобы размер шрифта корректно обновлялся

- Нормализовал ввод суммы: добавил parsedSum (trim + замена , на . + проверка Number.isFinite), чтобы одинаково обрабатывались значения с запятой и точкой.

- Переписал вычисления поиска комбинации: found и recount теперь работают от parsedSum

- Добавил hasInput с trim() для корректной обработки пустого/пробельного ввода.

- Реализовал тултипы для выведения информации о выплатах (исправил на корректный вариант, в том числе разобрался с иконкой)

**Дальше ждем дизайн и продуктовую постановку**
